src/sentry/ingest/consumer/processors.py

```diff
@@ -187,6 +187,8 @@ def process_event(
                 event_id=event_id,
                 project_id=project_id,
             )
+            collect_span_metrics(project, data)
```

Comment:
Can we move this into `save_event_transaction`, and / or wrap it in a try / catch?

This line is part of a `try` block, but it only DLQs `KeyErrors`. Any other failure might block the consumer and cause a P0 incident.

https://github.com/getsentry/sentry/blob/503314b8deabd129e94599c5f3fb36b127d0da6a/src/sentry/ingest/consumer/processors.py#L141-L142

Reply:
Wrap in a try/catch is a good idea, sure 👍